### PR TITLE
rendervulkan: Fix release barrier

### LINKED ISSUE #338: Dies on startup with debug radv
This barrier:

```
	bool useForeignQueue = !BIsNested() && g_vulkanSupportsModifiers;

	/* The non-foreign path is very hinky, and for the foreign path the acquire
	 * barriers are still missing (but are unlikely to do anything on radv). Also
	 * missing is the initial pipeline barrier from VK_IMAGE_LAYOUT_UNDEFINED. */
	VkImageMemoryBarrier memoryBarrier =
	{
		.sType = VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER,
		.srcAccessMask = pScreenshotTexture ? VK_ACCESS_TRANSFER_READ_BIT : VK_ACCESS_SHADER_WRITE_BIT,
		.dstAccessMask = useForeignQueue ? (VkAccessFlagBits)0 : VK_ACCESS_MEMORY_READ_BIT,
		.oldLayout = VK_IMAGE_LAYOUT_GENERAL,
		.newLayout = VK_IMAGE_LAYOUT_GENERAL,
		.srcQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED,
		.dstQueueFamilyIndex = useForeignQueue ? VK_QUEUE_FAMILY_FOREIGN_EXT
						       : VK_QUEUE_FAMILY_IGNORED,
		.image = compositeImage,
		.subresourceRange = subResRange
	};
	
	vkCmdPipelineBarrier( curCommandBuffer, pScreenshotTexture ? VK_PIPELINE_STAGE_TRANSFER_BIT : VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT,
			      useForeignQueue ? 0 : VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,
			      0, 0, nullptr, 0, nullptr, 1, &memoryBarrier );
```

Blows up in radv with this assert:

```
assert(src_family == cmd_buffer->queue_family_index ||
             dst_family == cmd_buffer->queue_family_index);
```

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Need to use our queueFamilyIndex as the src.

Closes #338

### CODE REVIEW

File: src/rendervulkan.cpp [hunk 1]
```diff
@@ -2352,9 +2352,12 @@ bool vulkan_composite( struct Composite_t *pComposite, struct VulkanPipeline_t *
 		.dstAccessMask = useForeignQueue ? (VkAccessFlagBits)0 : VK_ACCESS_MEMORY_READ_BIT,
 		.oldLayout = VK_IMAGE_LAYOUT_GENERAL,
 		.newLayout = VK_IMAGE_LAYOUT_GENERAL,
-		.srcQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED,
```
Comment (BNieuwenhuizen) [inline src/rendervulkan.cpp:2355]: Actually if we create with exclusive, we never should use `VK_QUEUE_FAMILY_IGNORED` so we can always set `srcQueueFamilyIndex` to `queueFamilyIndex` and the else part of `dstQueueFamilyIndex` to `queueFamilyIndex` too.

Sorry for being too lazy the first time ...

Comment (Joshua-Ashton) [reply src/rendervulkan.cpp:2355]: Updated.

**Comment by BNieuwenhuizen** [comment]
LGTM

**Comment by emersion** [comment]
Thanks!

**Pull request merged**